pkg/api/oidc_login_handler.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+
+	"github.com/gorilla/sessions"
+	"github.com/treeverse/lakefs/pkg/logging"
+	"golang.org/x/oauth2"
+)
+
+const (
+	OIDCAuthSessionName = "auth_session"
+
+	IDTokenClaimsSessionKey = "id_token_claims"
+	StateSessionKey         = "state"
+
+	stateByteSize = 32
```

Comment:
Is there a recommendation for this?  The best I could find was [this on login.gov](https://developers.login.gov/oidc/):

> **state**
> A unique value at least 22 characters in length used for maintaining state between the request and the callback. This value will be returned to the client on a successful authorization.

Admittedly this is quite a weak recommendation.

22 characters from a set of (say) 64 different characters is 6*22=132 bites, or 32.5 bytes.  But I would be happier to specify this in terms of characters, and use 22 or some recommended value.  nanoid is your friend for doing this sort of thing.

Reply:
Done